src/Blazored.Typeahead/wwwroot/blazored-typeahead.js

```diff
@@ -16,5 +18,64 @@ window.blazoredTypeahead = {
                 }
             });
         }
+    },
+    onOutsideClickClear: function (element) {
+
+        if (element == null) {
+            return;
+        }
+
+        var bId = ""; 
+        for (var clearCount = 0; clearCount < element.attributes.length; clearCount++) {
+            var a = element.attributes[clearCount];
+            if (a.name[0] == '_') {
+                bId = a.name;
+                break;
+            }
+        }
+
+        var func = onOutsideClickFunctions[bId];
+        if (func == null || func == "undefined") {
+            return;
+        }
+        window.removeEventListener("click", func);
+        onOutsideClickFunctions[bId] = null;
+    },
+    onOutsideClick: function (searchTextElement, dotnetRef, methodName, clearOnFire) {
+
+        if (searchTextElement == null) {
+            return;
+        }
+
+        var bId = "";//get the blazor internal ID to distinguish different components
+        for (var clearCount = 0; clearCount < searchTextElement.attributes.length; clearCount++) {
+            var a = searchTextElement.attributes[clearCount];
+            if (a.name[0] == '_') {

```

Comment:
Nit pick: Should this be `_bl_` as that is the full prefix Blazor uses for internal IDs?

Reply:
Yep good point, I just have an aversion to js startsWith for compatibility issues but I think that's an aversion I can now start to let go ^^ - this is included in the new commit